docs/source/mistral_jinja.rst

```diff
@@ -0,0 +1,176 @@
+that you can do with Jinja:
+
+* Select values for a key from a list of dictionary.
+* Simple arithmetic.
+* Evaluation of boolean logic.
+* Use conditional logic and builtin filters to evaluate and transform data.
+
+.. note::
+
+    Please refer to offical documentation for Mistral and Jinja. The documentation here
+    is meant to help |st2| users get a quick start.
+
+Basics
+++++++
+The following are statements in the workflow and task definition that accepts Jinja:
+
+* task action input
+* task concurrency
+* task on-complete
+* task on-error
+* task on-success
+* task pause-before
+* task publish
+* task retry break-on
+* task retry continue-on
+* task retry count
+* task retry delay
+* task timeout
+* task wait-before
+* task wait-after
+* task with-items
+* workflow output
+
+Each of the statements can take a string with one or more Jinja expressions. Each expression in the
+string should be encapsulated with ``{{ }}``. Code block using ``{% %}`` is also supported. Please
+note that the symbols ``{`` and ``}`` will conflict with JSON and Jinja expressions must always be
+encapsulated with quotes or double quotes in the workflow definition.
+
+.. note::
+
+    Mixing of both YAQL and Jinja expressions in a single statement is not supported.
+
+When evaluating a Jinja expression, Mistral also passes a JSON dictionary (aka context) to the
+Jinja templating engine. The context contains all the workflow inputs, published variables, and
+result of completed tasks up to this point of workflow execution including the current task. The
+Jinja expression can refer to one or more variables in the context. The reserved symbol ``_`` is
+used to reference the context. For example, given the context ``{"ip": "127.0.0.1", "port": 8080}``,
+the string ``https://{{ _.ip }}:{{ _.port }}/api`` returns ``https://127.0.0.1:8080/api``. The
+following is the same example used in a workflow:
+
+.. code-block:: yaml
+
+    version: '2.0'
+
+    examples.yaql-basic:
+        type: direct
+        input:
+            - ip
+            - port
+        tasks:
+            task1:
+                action: examples.call-api
+                input:
+                    endpoint: https://{{ _.ip }}:{{ _.port }}/api
+
+.. note::
+
+    The reserved symbol to reference the workflow context for Jinja is different than YAQL
+    expressions. This is due to the differences in the limitation between the two engines.
+    As mentioned above, the symbol "_" is used to access context in Jinja whereas the symbol
+    "$" is used in YAQL.
+
+The following is a more complex workbook example with a few more Jinja expressions. There are
+variables passed to input parameters and begin published after task completion. Please take
+note of the ``install_apps`` task in the ``configure_vm`` workflow. The input parameter ``cmd``
+is given the value formatted by the Jinja for loop. Unlike YAQL, string in a Jinja expression
```

Comment:
Should be either "Unlike YAQL, *a* string in a Jinja expression..." or "Unlike YAQL, *the* string in a Jinja expression..."